# ğŸ“Š Full RNA-seq Analysis in Yeast

## ğŸ§¬ Project Overview
Re-analysis of 24 yeast RNA-seq samples to investigate stabilization of snRNAs upon RRP6 deletion. Built using Snakemake and R for a reproducible and modular pipeline.

## ğŸ“ Data Source
- GEO Accession: [GSE135056](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE135056)
- Original Study: [PMID: 32187185](https://www.ncbi.nlm.nih.gov/pubmed/32187185)
- Re-analysis co-authored publication: [Microbial Cell Publication 2024](https://doi.org/10.15698/mic2024.05.823)

## ğŸ”§ Pipeline Structure
- Raw data â†’ QC â†’ Trimming â†’ Alignment â†’ Counting (Automated via Snakemake)
- Downstream analysis via DESeq2, PCA, volcano plots, heatmaps (R)

## âš™ï¸ Tools & Dependencies
- Snakemake â‰¥7.0  
- HISAT2, SAMtools, FastQC, featureCounts  
- R â‰¥4.2 with DESeq2, ggrepel, pheatmap  
- Conda env spec: [`rnaseq_env.yaml`](/envs/rnaseq_env.yaml)

## ğŸ“¦ Directory Structure
<pre>
  .
â”œâ”€â”€ .DS_Store
â”œâ”€â”€ .Rhistory
â”œâ”€â”€ .snakemake
â”œâ”€â”€ alignment
â”œâ”€â”€ benchmarks
â”œâ”€â”€ config
â”œâ”€â”€ counts
â”œâ”€â”€ csvs
â”œâ”€â”€ dir_structure.txt
â”œâ”€â”€ envs
â”œâ”€â”€ figures
â”œâ”€â”€ logs
â”œâ”€â”€ quality-check
â”œâ”€â”€ raw_data
â”œâ”€â”€ reference
â”œâ”€â”€ reports
â”œâ”€â”€ scripts
â”œâ”€â”€ Snakefile
â””â”€â”€ trimmed
</pre>

16 directories, 4 files


## ğŸš€ How to Run
1. Clone repo and create conda env:
   ```bash
   conda env create -f envs/rnaseq_env.yaml

## ğŸ§  Biological Insight & Sample Curation

This re-analysis obtains the raw reads for the WT and *rrp6*-Î” samples from the original study by Victorino et al., 2020.

The primary objective of this study was to confirm the wet-lab hypothesis that **RRP6p** is responsible for degrading mature snRNAs in Baker's yeast *Saccharomyces cerevisiae*. This is an overly simplified version of the actual problem which is detailed in the publication Chaudhuri et al., 2024. Upon analysing the full dataset of 24 samples, we obtained volcano plots and heatmaps that clearly showed a significant upregulation of most (58/81) snRNAs in yeast, which is exactly what was predicted from our experiments.

During exploratory PCA of variance-stabilized counts, we identified one **biological replicate in both WT and rrp6 knockout conditions** to show **~22% variation** compared to its group centroid. To ensure robust differential expression results, this outlier was excluded prior to running DESeq2 once again.

Both the analyses scripts are included in the original along with the removal of the outliers. The PCA plots from before and after the removal of outliers are provided below.

The outliers were identified in 2 modified PCA plots, with one plot having a standard approach with labels while another with a more colourful approach to identify the replicates.

### ğŸ”¬ PCA plot (before removing outlier)
![PCA Plot](figures/PCA_with_outliers.jpeg)

### ğŸ”¬ PCA plot (after removing outlier)
![PCA Plot](figures/PCA_3_reps.jpeg)

### ğŸ”¬ PCA plot (identifying by labels)
![PCA Plot](figures/PCA_labelled_reps.jpeg)

### ğŸ”¬ PCA plot (identifying by colour)
![PCA Plot](figures/PCA_identify_reps.jpeg)

### ğŸŒ‹ Volcano plot highlighting snRNA stabilization
![Volcano Plot](figures/Volcano_snR_only.jpeg)

This approach preserved biological signal while ensuring technical reproducibility. The snRNA specific volcano plot revealed that majority of the snRNAs were upregulated significantly, thereby reinforcing the hypothesis that **RRP6 actively degrades mature snRNAs under normal conditions**.

